pytensor/link/jax/dispatch/slinalg.py

```diff
@@ -54,7 +54,21 @@ def solve(a, b):
             dl = jax.numpy.diagonal(a, offset=-1, axis1=-2, axis2=-1)
             d = jax.numpy.diagonal(a, offset=0, axis1=-2, axis2=-1)
             du = jax.numpy.diagonal(a, offset=1, axis1=-2, axis2=-1)
-            return jax.lax.linalg.tridiagonal_solve(dl, d, du, b, lower=lower)
+            # jax requires dl and du to have the same shape as d
+            dl = jax.numpy.pad(dl, (1, 0))
+            du = jax.numpy.pad(du, (0, 1))
+            # if b is a vector, broadcast it to be a matrix
+            b_is_vec = len(b.shape) == 1
```

Comment:
You don't need to check this at runtime. The `Solve` `Op` has a property `b_ndim`, so you can do:

```py
b_is_vec = op.b_ndim

if assume_a == 'tridiagonal':
    ... # carry on
```

Reply:
The check as written will also fail in the batched case (that's why we have it at the Op level)

Reply:
awesome

Reply:
you'd have to do `b_is_vec = op.b_ndim == 1` though, no? because `bool(op.b_ndim) -> True` for `b_ndim > 0`

Reply:
yes exactly, my code has an error